scout/build/case.py

```diff
@@ -115,52 +115,93 @@ def build_case(case_data, adapter):
+
+    set_analysis_date(case_obj, case_data)
+    set_assignee(case_obj, case_data)
+    set_causatives(case_obj, case_data)
+    set_suspects(case_obj, case_data)
+    set_timestamps(case_obj)
+
+    case_obj["is_research"] = False
+    case_obj["lims_id"] = case_data.get("lims_id", "")
+    case_obj["rerun_requested"] = False
+    case_obj["research_requested"] = False
+    case_obj["status"] = "inactive"
+    case_obj["panels"] = get_panels(case_data, adapter)
+    case_obj["dynamic_gene_list"] = []
+
+    # Meta data
+    set_genome_build(case_obj, case_data)
+
+    if case_data.get("rank_model_version"):
+        case_obj["rank_model_version"] = str(case_data["rank_model_version"])
 
-    # Owner allways has to be part of collaborators
+    if case_data.get("sv_rank_model_version"):
+        case_obj["sv_rank_model_version"] = str(case_data["sv_rank_model_version"])
+
+    if case_data.get("rank_score_threshold"):
+        case_obj["rank_score_threshold"] = float(case_data["rank_score_threshold"])
+
+    set_cohort(case_obj, case_data)
+    sync_cohort_view(case_obj, institute_obj, adapter)
+    set_phenotype_terms(case_obj, case_data, adapter)
+    set_phenotype_groups(case_obj, case_data, adapter)
+
+    # Files
+    case_obj["madeline_info"] = case_data.get("madeline_info")
+
+    case_obj["custom_images"] = case_data.get("custom_images")
+    set_custom_report(case_obj, case_data)
+    case_obj["vcf_files"] = case_data.get("vcf_files", {})
+    case_obj["delivery_report"] = case_data.get("delivery_report")
+
+    case_obj["has_svvariants"] = False
+    if case_obj["vcf_files"].get("vcf_sv") or case_obj["vcf_files"].get(
+        "vcf_sv_research"
+    ):
+        case_obj["has_svvariants"] = True
+
+    case_obj["has_strvariants"] = False
+    if case_obj["vcf_files"].get("vcf_str"):
+        case_obj["has_strvariants"] = True
+
+    case_obj["is_migrated"] = False
+
+    # What experiment is used, alternatives are rare (rare disease) or cancer
+    case_obj["track"] = case_data.get("track", "rare")
+    case_obj["group"] = case_data.get("group", [])
+
+    return case_obj
+
+
+def get_collaborators(case_data):
+    """Return a list of collaborators with owner appended.  Owner must
+    be part of collaborators"""
     collaborators = set(case_data.get("collaborators", []))
     collaborators.add(case_data["owner"])
-    case_obj["collaborators"] = list(collaborators)
+    return list(collaborators)
 
-    if case_data.get("assignee"):
-        case_obj["assignees"] = [case_data["assignee"]]
 
-    case_obj["smn_tsv"] = case_data.get("smn_tsv")
-
-    # Individuals
+def get_individuals(case_data):
+    """Returns samples with with affected individual first"""
     ind_objs = []
     try:
         for individual in case_data.get("individuals", []):
             ind_objs.append(build_individual(individual))
     except Exception as error:
-        ## TODO add some action here
+        # TODO: Do something useful here
```

Comment:
I think a reasonable exception should be thrown. But which one? Either `ConfigError("build individuals fails")`. But I would prefer not to catch the error -hiding the original fault- and let it crash. That is remove the `try-catch`. Because if there is an error here, it makes no sense to continue execution.

Reply:
Fixed!